### PR TITLE
fix: Delay checking for __coverage__ until helpers run because __coverage__ is undefined when Jest starts

### PR DESCRIPTION
**Pull request by vkarpov15**
Re: temporalio/temporal-pendulum#21

## What was changed

Don't check for `__coverage__` when the module loads, but when the `augment()` methods run.

## Why?

Jest quirks: `__coverage__` isn't set when the module first loads.

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: packages/nyc-test-coverage/src/index.ts [hunk 1]
```diff
@@ -5,13 +5,15 @@ import libCoverage from 'istanbul-lib-coverage';
...
-// Check if running through nyc or some other Istanbul-based tool.
-// If not, any `workflowCoverage()` tools are a no-op.
-const hasCoverageGlobal = '__coverage__' in global;
-
 export class WorkflowCoverage {
   coverageMap = libCoverage.createCoverageMap();
 
+  // Check if running through nyc or some other Istanbul-based tool.
+  // If not, any `workflowCoverage()` tools are a no-op.
+  hasCoverageGlobal() {
```
Comment (bergundy) [inline packages/nyc-test-coverage/src/index.ts:13]: Make this private

File: packages/nyc-test-coverage/src/index.ts [hunk 2]
```diff
@@ -23,7 +25,7 @@ export class WorkflowCoverage {
       throw new TypeError('Cannot automatically instrument coverage without specifying `workflowsPath`');
     }
 
-    if (!hasCoverageGlobal) {
+    if (!this.hasCoverageGlobal()) {
```
Comment (bergundy) [inline packages/nyc-test-coverage/src/index.ts:28]: Why is this necessary?

Comment (vkarpov15) [reply packages/nyc-test-coverage/src/index.ts:28]: This avoids instrumenting Workflow code if coverage isn't enabled. I think it is beneficial to avoid piping the Workflow code through istanbul-instrumenter-loader if coverage isn't enabled. I don't expect it to cause any problems, but I'd prefer to err on the side of caution there.

Comment (bergundy) [reply packages/nyc-test-coverage/src/index.ts:28]: I'd throw an error here instead then instead of silently failing.

Comment (vkarpov15) [reply packages/nyc-test-coverage/src/index.ts:28]: Do you want to put the responsibility on the user to avoid calling this function if code coverage is disabled?

Comment (bergundy) [reply packages/nyc-test-coverage/src/index.ts:28]: Do you expect this to be a common scenario?

Comment (joebowbeer) [reply packages/nyc-test-coverage/src/index.ts:28]: I think this is very common, IIUC. The user doesn't want to have to rip out their workflowCoverage augmentation code just because they disabled coverage on a specific test run.

Comment (bergundy) [reply packages/nyc-test-coverage/src/index.ts:28]: SGTM, please ignore this comment then.
I would maybe consider issuing a warning to let the user know that coverage is disabled.

Comment (vkarpov15) [reply packages/nyc-test-coverage/src/index.ts:28]: I agree with @joebowbeer . It isn't uncommon to run tests with coverage on vs coverage off. And that means every user would have to implement their own check for whether they're running with coverage or not.

**Review by bergundy — APPROVED** [review]
